### PR TITLE
meta: add `GOVERNANCE.md`

### PR DESCRIPTION
**Pull request by ErichDonGubler**

### CODE REVIEW

**Comment by ErichDonGubler** [comment]
Currently in draft to reflect that I would like feedback from decision-making groups identified in this current revision. To wit: @gfx-rs/wgpu, @gfx-rs/naga

I've given my best first approximation of how governance currently works in the WGPU project, but I expect that there's quite a bit I'm missing, or that's not quite correct.

File: GOVERNANCE.md
Comment (cwfitzgerald) [inline GOVERNANCE.md]: Let's call this community leadership and have me, wumpf, groves

File: GOVERNANCE.md
Comment (cwfitzgerald) [inline GOVERNANCE.md:41]: We should mention deno and crowlkats

File: GOVERNANCE.md
Comment (cwfitzgerald) [inline GOVERNANCE.md]: This should be at the end - this is legaleze no user cares about.

File: GOVERNANCE.md
Comment (ErichDonGubler) [reply GOVERNANCE.md:41]: I was careful to avoid mentioning groups smaller than Firefox (because I don't want details that are prone to changing in a document that seems frankly unlikely to get updated often), but if you'd like them explicitly mentioned, then SGTM!

File: GOVERNANCE.md
Comment (ErichDonGubler) [reply GOVERNANCE.md]: I think that having it at the front is the only way it'll ever be seen, but 🤷🏻‍♂️ I don't have particularly strong feelings about the position.

Comment (cwfitzgerald) [reply GOVERNANCE.md]: This strikes me as the kind of thing that exists so we can point to when someone tries to be an ass and use this document against us.

File: GOVERNANCE.md
Comment (cwfitzgerald) [reply GOVERNANCE.md:41]: We have an explicitly special relationship with deno (we host deno_webgpu!) which has existed for a few years, I don't expect this to really change. We could also put him under the community banner, as I think of him as part of community interest.

File: GOVERNANCE.md
Comment (ErichDonGubler) [reply GOVERNANCE.md]: I think your analysis is on point. 🙂

File: GOVERNANCE.md
Comment (Wumpf) [inline GOVERNANCE.md]:
```suggestion
enjoyed much contribution and use from other projects that require graphics
```

File: GOVERNANCE.md
Comment (Wumpf) [inline GOVERNANCE.md]: Very happy with the first point, but I feel a few important aspects of wgpu are not quite captured yet here:
* seamless use as a graphics backend in applications targeting **both native & web**
* provide a WebGPU _style_ graphics API abstraction to native _Rust_ applications 

currently 'native' is only under the "platform of experimentation" which sounds a bit like anything that isn't part of a WebGPU implementation isn't fully supported.
I'm still pondering a bit how much our many many native extensions are directly part of the mission: they're clearly in part (!) experimentation but realistically this is far outside of what can be expected to be part of WebGPU any time soon. But these are still important pieces in providing a flexible WebGPU-like api to native Rust users, so I believe this deserves a place here.
My understanding of wgpu's mission is first WebGPU impl (as already well described above), then native graphics abstraction library with seamless web support, then platform for experimentation as described. All of these are obviously interconnected (well otherwise we wouldn't put it into a single family of libraries), but they do occasionally pull into different directions and it's worth pointing them and the implicit hierarchy out

File: GOVERNANCE.md
Comment (nical) [inline GOVERNANCE.md]: typo: regulrly -> regularly

File: GOVERNANCE.md
Comment (ErichDonGubler) [reply GOVERNANCE.md]: Implemented with fe1d09d0a.

File: GOVERNANCE.md
Comment (ErichDonGubler) [reply GOVERNANCE.md:41]: Noted Deno and @crowlKats with a62303575.

I've added Firefox contributor names with a62303575 (minus @bradwerth, unless he also wants to be on there), though I think maybe that's not a good idea for us, since membership and contribution can fluctuate over time? CC @jimblandy, @teoxoy.

File: GOVERNANCE.md
Comment (ErichDonGubler) [reply GOVERNANCE.md]: Moved to the bottom with 465c75ddd.

File: GOVERNANCE.md
Comment (ErichDonGubler) [reply GOVERNANCE.md]: D'oh, implemented with 465c75ddd.

File: GOVERNANCE.md
Comment (ErichDonGubler) [reply GOVERNANCE.md]: D'oh, fixed with b2e21e072.

File: GOVERNANCE.md
Comment (ErichDonGubler) [reply GOVERNANCE.md:41]: Resolving for now, but feel free to un-resolve if somebody has an objection to the current contents.

File: GOVERNANCE.md
Comment (ErichDonGubler) [reply GOVERNANCE.md]: I've attempted a revision of this text along the lines of what it seems you want at cd8cf89d5, but I'm not confident I've captured things to your satisfaction. Whaddaya think?

File: GOVERNANCE.md
Comment (cwfitzgerald) [inline GOVERNANCE.md]: `/s/...is/...are` subject-verb agreement

File: GOVERNANCE.md
Comment (cwfitzgerald) [inline GOVERNANCE.md]: I will take mild offense on nical's behalf, he definitely belongs on this list :)

File: GOVERNANCE.md
Comment (kpreid) [inline GOVERNANCE.md]: The word “ecosystem” seems wrong to me. In particular, to me it implies _at least_ a group of _separately administrated_ interrelated projects, whereas this document refers to a _uniformly_ administrated set.

How about something narrower and plainer like “The WGPU project is a collection of open-source libraries that…”

File: GOVERNANCE.md
Comment (kpreid) [inline GOVERNANCE.md]:
```suggestion
NOTE: The above is a snapshot of a perpetually changing state of affairs in
```

File: GOVERNANCE.md
Comment (Wumpf) [reply GOVERNANCE.md]: nice, has all the points baked in, I like it. 🚢

**Review by Wumpf — APPROVED** [review]
there's still good feedback coming in that needs to be incorporated as of writing, but on my side I'm quite happy with this otherwise. Thanks again!

File: GOVERNANCE.md
Comment (cwfitzgerald) [reply GOVERNANCE.md]:
```suggestion
    1. …are correct and fully conformant.
    1. …are portable across all major platforms, that is, …
        1. …`wgpu-core` enables JavaScript platforms to implement their own
           proper WebGPU API.
        1. …`wgpu` provides a WebGPU-style API library for native applications,
           which allows shipping to all major platforms, including WebGPU's
           JavaScript API.
    1. …are performant enough to enable demanding applications.
```

In useful suggestion form, as opposed to non-useful unix-nerd form.

File: GOVERNANCE.md
Comment (ErichDonGubler) [reply GOVERNANCE.md]: Derp, I had actually thought to make this edit, but dropped it mentally, for some reason. Thanks!

File: GOVERNANCE.md
Comment (ErichDonGubler) [reply GOVERNANCE.md]: ofrick urite sorry @nical 😭

This is why I didn't wanna name people, yo.

Will fix soon.

Comment (ErichDonGubler) [reply GOVERNANCE.md]: Fixed with 598bc0ce9.

File: GOVERNANCE.md
Comment (ErichDonGubler) [reply GOVERNANCE.md]: 🤦🏻‍♂️Fixed with 76baf1055. That's what I get for picking a form that's requires fuller re-reading to catch, _and_ making two separate edits without double-checking the entire chain of grammar.

File: GOVERNANCE.md
Comment (ErichDonGubler) [reply GOVERNANCE.md]: Fixed with d6a8c81c8.

File: GOVERNANCE.md
Comment (ErichDonGubler) [reply GOVERNANCE.md]: Hmm. It took me some thinking about this deeply, but I think you're right. That's an interesting distinction I hadn't considered before.

Is WGPU not an ecosystem, though? There are several third-party crates that integrate directly with `wgpu` and `naga`. In that case, maybe rephrasing to "…_creates_ an ecosystem that…" (emphasis on edit, not to be propagated in final changes).

Comment (cwfitzgerald) [reply GOVERNANCE.md]: Yeah, i would say it _has_ an ecosystem, but is not one itself

Comment (kpreid) [reply GOVERNANCE.md]: > Is WGPU not an ecosystem, though?

Well, I would understand “the WGPU ecosystem” as the WGPU project plus those third-party crates. But “third-party” here means, at a minimum, “not governed by the WGPU project organization” — and to me, “ecosystem” further implies that those parties created the crates _by their own choices, to serve their own goals_ and as a consequence made WGPU more useful for everyone.

This document describes governance for the WGPU project, not things that are not the WGPU project. Governing the “ecosystem” by these principles and leadership is a logical contradiction, because if you could, it would cease to be an “ecosystem” and become part of the project.

Comment (ErichDonGubler) [reply GOVERNANCE.md]: Changed as suggested with 7a73c1622.

**Comment by ErichDonGubler** [comment]
I think the only feedback I'd like to get before merging is from individuals mentioned as influential members of the community in this new document. IOW, **I'd like to see if we can get a nice green checkmark from**:

* @cwfitzgerald
* @grovesNL
* @jimblandy
* @nical
* @teoxoy
* @crowlKats

I intend to merge this by end-of-week, or when the above individuals approve (whichever is sooner). I don't see a good reason to block for longer than that.

**Review by teoxoy — APPROVED** [review]

File: GOVERNANCE.md
Comment (cwfitzgerald) [inline GOVERNANCE.md]:
```suggestion
authors to write portable and performant graphics programs_. It was originally
conceived to provide an implementation of WebGPU for Firefox, and 
settled into something that could be shipped on all web browsers.
```

The original feels a bit run-on-ey

**Review by cwfitzgerald — APPROVED** [review]
Approved with nit!

File: GOVERNANCE.md
Comment (ErichDonGubler) [reply GOVERNANCE.md]: Implemented as suggested with e71c68132.

**Review by crowlKats — APPROVED** [review]

**Pull request merged**
